library/src/interactor_impl.cxx

```diff
@@ -500,6 +456,66 @@ class interactor_impl::internals
     this->Window.render();
   }
 
+  //----------------------------------------------------------------------------
+  void StartEventLoop(double deltaTime, std::function<void()> userCallBack)
+  {
+    // Copy user callback
+    this->EventLoopUserCallBack = std::move(userCallBack);
+
+    // Configure UI delta time
+    vtkRenderWindow* renWin = this->Window.GetRenderWindow();
+    vtkF3DRenderer* ren = vtkF3DRenderer::SafeDownCast(renWin->GetRenderers()->GetFirstRenderer());
+    ren->SetUIDeltaTime(deltaTime);
+
+    // Configure animation delta time
+    this->AnimationManager->SetDeltaTime(deltaTime);
+
+    // Create the timer
+    this->EventLoopTimerId = this->VTKInteractor->CreateRepeatingTimer(deltaTime * 1000);
+
+    // Create the callback and add an observer
+    vtkNew<vtkCallbackCommand> timerCallBack;
+    timerCallBack->SetCallback(
+      [](vtkObject*, unsigned long, void* clientData, void*)
+      {
+        std::function<void()>* callBackPtr = static_cast<std::function<void()>*>(clientData);
+        (*callBackPtr)();
+      });
+    this->EventLoopObserverId =
+      this->VTKInteractor->AddObserver(vtkCommand::TimerEvent, timerCallBack);
+    timerCallBack->SetClientData(&this->EventLoopCallBack);
```

Comment:
```suggestion
    vtkNew<vtkCallbackCommand> timerCallBack;
    timerCallBack->SetCallback(
      [](vtkObject*, unsigned long, void* clientData, void*)
      {
        internals* that = static_cast<internals*>(clientData);
        that->EventLoop();
      });
    this->EventLoopObserverId =
      this->VTKInteractor->AddObserver(vtkCommand::TimerEvent, timerCallBack);
    timerCallBack->SetClientData(this);
```

and ditch `EventLoopCallBack`

Reply:
Im quite sure that doesnt work because the delta lifetime would not extend after the scope, this is why we had the whole map for storing callbacks before.

I can try your code if you insists though, let me know

Reply:
Just read the code again and I don't get why it wouldnt work

Reply:
indeed, it works.